django/thunderstore/account/tests/test_user_flags.py

```diff
@@ -0,0 +1,166 @@
+from datetime import timedelta
+
+import pytest
+from django.contrib.auth.models import AnonymousUser
+from django.utils import timezone
+
+from thunderstore.account.factories import UserFlagFactory, UserFlagMembershipFactory
+from thunderstore.account.models import UserFlag
+from thunderstore.core.factories import UserFactory
+from thunderstore.core.types import UserType
+
+
+@pytest.mark.django_db
+def test_user_flags_empty_for_unauthenticated(
+    user_flag: UserFlag,
+):
+    start_time = timezone.now()
+    late_time = timezone.now() + timedelta(days=10)
+
+    UserFlagMembershipFactory(
+        flag=user_flag,
+        datetime_valid_from=start_time,
+        datetime_valid_until=None,
+    )
+
+    assert UserFlag.get_active_flags_on_user(None, late_time) == []
+    assert UserFlag.get_active_flags_on_user(AnonymousUser(), late_time) == []
+
+
+@pytest.mark.django_db
+def test_user_flags_overlapping_memberships_return_distinct(
+    user: UserType,
+    user_flag: UserFlag,
+):
+    early_time = timezone.now()
+    start_time = early_time + timedelta(seconds=1)
+    late_time = start_time + timedelta(days=100000)
+
+    for _ in range(3):
+        UserFlagMembershipFactory(
+            user=user,
+            flag=user_flag,
+            datetime_valid_from=start_time,
+            datetime_valid_until=None,
+        )
+
+    assert UserFlag.get_active_flags_on_user(user, early_time) == []
+    assert UserFlag.get_active_flags_on_user(user, late_time) == [user_flag.identifier]
+    assert UserFlag.get_active_flags_on_user(None, late_time) == []
+
+
+@pytest.mark.django_db
+def test_user_flags_without_end_time_are_active(
+    user: UserType,
+    user_flag: UserFlag,
+):
+    early_time = timezone.now()
+    start_time = early_time + timedelta(seconds=1)
+    late_time = start_time + timedelta(days=100000)
+
+    UserFlagMembershipFactory(
+        user=user,
+        flag=user_flag,
+        datetime_valid_from=start_time,
+        datetime_valid_until=None,
+    )
+    assert UserFlag.get_active_flags_on_user(user, early_time) == []
+    assert UserFlag.get_active_flags_on_user(user, late_time) == [user_flag.identifier]
+    assert UserFlag.get_active_flags_on_user(None, late_time) == []
```

Comment:
I would yeet this assertion as irrelevant and do one for testing with `start_time` to bring an edge case into play instead.